lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
```

Comment:
Gotcha. I would just say that we have to make sure the behavior from the old stays the same after this change. For example, `[1,2,3]` ends up very different in the SQL than `[[1,2,3]]`, so we don't wan to accidentally start not flattening some of the arguments with the change.